gpflow/posterior.py

```diff
@@ -315,6 +318,10 @@ def _precompute(self):
 
         Qinv = Kuu.inverse() - KuuInv_covu_KuuInv  # XXX LinearOperator does not support `-`
 
+        tf.debugging.assert_shapes(
```

Comment:
I have added this to the `LinearOperatorBasePosterior`. Since this is marked as WIP in the docstring, could this class be removed from this PR and handled later?

Reply:
Yes, I've moved it out into this branch: https://github.com/GPflow/GPflow/pull/new/st/posterior_with_linear_operators - feel free to push this change directly to that branch.